src/Identicon/Generator/BaseGenerator.php

```diff
@@ -44,6 +44,8 @@ class BaseGenerator
      */
     private $arrayOfSquare = array();
 
+    abstract public function getMimeType();
```

Comment:
This method is already inside the Interface. I think we do not need this class to be abstract.